stable_baselines/common/vec_env/base_vec_env.py

```diff
@@ -83,6 +83,42 @@ def close(self):
         """
         pass
 
+    @abstractmethod
+    def get_attr(self, attr_name, indices=None):
+        """
+        Provides a mechanism for getting class attributes from vectorized environments
+
+        :param attr_name: (str) The name of the attribute whose value to return
+        :param indices: (list,int) Indices of envs to get attribute from
+        :return: (list) List of values of 'attr_name' in all environments
+        """
+        pass
+
+    @abstractmethod
+    def set_attr(self, attr_name, value, indices=None):
+        """
+        Provides a mechanism for setting arbitrary class attributes inside vectorized environments
+
+        :param attr_name: (str) Name of attribute to assign new value
+        :param value: (obj) Value to assign to 'attr_name'
+        :param indices: (list,int) Indices of envs to assign value
+        :return: (list) in case env access methods might return something, they will be returned in a list
+        """
+        pass
+
+    @abstractmethod
+    def env_method(self, method_name, indices=None, *method_args, **method_kwargs):
```

Comment:
How about changing signature to:
`def env_method(self, method_name, *method_args,  indices=None, **method_kwargs):`
This would make `indices` only accessible as a keyword-argument, but this seems maybe desirable, and certainly better than breaking the API for existing users of `env_method`.